client.go

```diff
@@ -23,67 +22,69 @@ var (
 		MaxTries:            6}
 )
 
+// A ResponseFuture is a container for a Response which will materialise at some point.
 type ResponseFuture struct {
-	cancel context.CancelFunc
-	done   <-chan struct{} // guards access to r
-	r      Response
+	done <-chan struct{} // guards access to r
+	r    Response
 }
 
+// WaitC returns a channel which can be waited upon until the response is available
 func (f *ResponseFuture) WaitC() <-chan struct{} {
 	return f.done
 }
 
+// Response provides access to the response object, blocking until it is available
 func (f *ResponseFuture) Response() Response {
 	<-f.WaitC()
 	return f.r
 }
 
-func (f *ResponseFuture) Cancel() {
```

Comment:
This is the only signature change which is not obvious (to me) if anyone will ever use this directly. We do *return* a ResonseFuture from SendVia (which is a public function) but whether a third party client would have code that attempts to call Cancel() isn't obvious.

Do we care or need to about breaking any protocol here? should this be a noop instead but still be supported?